File: app/src/main/java/com/mercandalli/tracker/push/PushModule.java
package com.mercandalli.tracker.push;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;

@Module
public class PushModule {

    private static final String CLOUD_MESSAGING_API_KEY = "AAAAy9qyE3M:APA91bEt8F9u7r_N8CL4mLvYUDPAJaAClu_GsiDavD0-7VmesKYGkSHx2jAPja8sSuCQ8QV4Q4Y1ZzAi8ykrdA7sQ2R_UhS6-BeAX8fzGrmafCeNcxZ9nfQ1HdGQJLJpDaTkRP_0by4c";

    @Singleton
    @Provides
    PushSenderManager providePushSenderManager(OkHttpClient okHttpClient) {
        return new PushSenderManagerNetwork(
                okHttpClient,
                CLOUD_MESSAGING_API_KEY);
    }
}

File: app/src/main/java/com/mercandalli/tracker/push/PushModule.kt
package com.mercandalli.tracker.push

import com.google.gson.Gson
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module
class PushModule {

    @Singleton
    @Provides
    internal fun providePushSenderManager(
            okHttpClient: OkHttpClient,
            gson: Gson): PushSenderManager {
        return PushSenderManagerNetwork(
                okHttpClient,
                gson,
                CLOUD_MESSAGING_API_KEY)
    }

    companion object {

        private val CLOUD_MESSAGING_API_KEY = "AAAAFnjUxws:APA91bGesReILlGlDKySySV0GWNxy4PMOboBdOB398p7Z-3o81A8a5l2E7fSgSz8u7ElJDqOOA4Kix_WBZndmSFJbadJmlC5gKQlRe5ChzK_FkdE4sgm9nOcdXtPLM5hFNd0K29pzJkV"
    }
}
